src/core/nginx.go

```diff
@@ -203,20 +210,30 @@ func (n *NginxBinaryType) GetNginxDetailsFromProcess(nginxProcess *Process) *pro
 		nginxDetailsFacade.ConfPath = path
 	}
 
-	stubStatusApiUrl, err := sdk.GetStubStatusApiUrl(nginxDetailsFacade.ConfPath, n.config.IgnoreDirectives)
-	if err != nil {
-		log.Tracef("Unable to get Stub Status API URL from the configuration: NGINX OSS metrics will be unavailable for this system. please configure aStub Status API to get NGINX OSS metrics: %v", err)
-	}
+	if len(n.statusUrls[nginxID]) == 0 || n.statusUrls[nginxID] == "" {
```

Comment:
i think the first check should be sufficient? they seems to be checking the same thing.

Reply:
also, potential race condition. The assignment for the `nginxID` in line 231 is mutex protected, but here it's not. I think a common pattern would be accessor functions that's mutex protected. Something like:

```
func (n ..) updateStatusURL(...) {
    n.statusUrlMutex.Lock()
    defer n.statusUrlMutex.Unlock(()
    ...
}
```

Reply:
there's no assignment here?

Reply:
updated to use the RLock